packages/tablev2/Util.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import { Icon } from "../icon";
+import { SystemIcons } from "../icons/dist/system-icons-enum";
+import * as dt from "../design-tokens/build/js/designTokens";
+import { css } from "emotion";
+
+///////////////////////////////////////////////////////////////////////////////
+//                                 DRAGGABLE                                 //
+///////////////////////////////////////////////////////////////////////////////
+
+let initialX: number = 0;
+export const className = css`
+  padding: 0 0.25em;
+  cursor: col-resize;
+  display: flex;
+  align-items: center;
+  height: 100%;
+  z-index: 10;
+  position: absolute;
+  top: 0;
+  right: 0;
+  background-color: ${dt.themeBgPrimary};
+  opacity: var(--draggable-opacity);
+  &:hover {
+    background-color: ${dt.themeBgHover};
+  }
+`;
+
+type E = React.MouseEvent;
+/**
+ * The draggable small handler used to size table columns.
+ */
+export const Draggable = (props: {
+  onRelativeXChange: (i: number) => void;
+}) => {
+  const [x, setX] = React.useState(0);
+
+  const updateMousePosition = (e: MouseEvent) => {
+    setX(e.clientX - initialX);
+  };
+
+  const onMouseUp = (e: MouseEvent) => {
+    props.onRelativeXChange(e.clientX - initialX);
+    setX(0);
+    document.removeEventListener("mousemove", updateMousePosition);
+    document.removeEventListener("mouseup", onMouseUp);
+  };
+  const onMouseDown = (e: E) => {
+    initialX = e.clientX;
+    document.addEventListener("mousemove", updateMousePosition);
+    document.addEventListener("mouseup", onMouseUp);
+  };
+
+  return (
+    <div
+      onMouseDown={onMouseDown}
+      className={className}
+      style={{ transform: `translate(${x}px, 0)` }}
+      role="button"
+    >
+      <div>
+        <Icon
+          shape={SystemIcons.ResizeHorizontal}
+          size={dt.iconSizeXxs}
+          color={dt.textColorSecondary}
+        />
+      </div>
+    </div>
+  );
+};
+
+///////////////////////////////////////////////////////////////////////////////
+//                                  SORTERS                                  //
+///////////////////////////////////////////////////////////////////////////////
+
+export type SorterFactory<T> = <B extends string, A extends Record<B, T>>(
+  prop: B
+) => Sorter<A>;
+
+const string: SorterFactory<string> = prop => dir => (a, b) =>
+  (dir === "asc" ? 1 : -1) * a[prop].localeCompare(b[prop]);
+
+const number: SorterFactory<number> = prop => dir => (a, b) =>
```

Comment:
@mperrotti today i tried integrating the table into dcos-ui. when writing custom sorters i was annoyed by having to check the direction which is either "asc" or "desc". i'm thinking of just passing a `1` instead of "asc" and a `-1` instead of "desc" so that the parameter can directly be multiplied:

```
const sortByNameWithGroupsAlwaysOnTop = dir => (a, b) =>
  a.__type === b.__type
    ? dir * strComp(a, b) // formerly: (dir === "asc" ? 1 : -1) * strComp(a, b)
    : +(a.__type === "group");
```

Reply:
I can't think of anything better than `dir` right now either 🤔

I'm a _little_ concerned that `1` and `-1` might not be as clear as `asc` and `desc`. However, I'm fine with that change if you and other reviewers think it's a better developer experience.

Reply:
I prefer "asc" and "desc", but maybe we can accept both?

Reply:
in `sortByNameWithGroupsAlwaysOnTop` how would you either have dir be `"asc" | "desc"` or `1 | -1`?
for the sake of clarity, i'd vow to not give consumers much freedom on how to use this.

Reply:
If I had to choose one, I would choose "asc" and "desc".